MVP for Extracting JS Scripts from HTML

dmorejon opened the pull request:
extractScripts.js parses the HTML test file and, for each set of <script> tags, creates a new .js file containing the Javascript.
test/testExtractScripts.js is the associated tests. 

This is the first step in the process of converting legacy HTML tests using js-test.js to use testharness.js.

dmorejon committed:
Extracts the JS  within <script> tags in HTML test.

The scripts are placed in a temporary directory, and each one is its own
.js file.

dmorejon committed:
Updated function documentation for extractScriptsFromHTML

dmorejon committed:
Added testExtractScripts.js, refactored extractScripts.js

dmorejon committed:
BROKEN: debugging temp directory empty file read

dmorejon committed:
Fixed async write issue, all tests pass.

dmorejon committed:
Stylistic changes and refactoring.

dmorejon committed:
ESLint setup and linter fixes. Moved files to src/ and test/

Hexcles commented on package.json:
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "webtesttransfomer",
+  "version": "1.0.0",
+  "description": "Ecosystem Infra tool to transform legacy tests using js-test.js to WPT tests using harnesstest.js",
```
```suggestion
  "description": "Ecosystem Infra tool to transform legacy tests using js-test.js to WPT tests using testharness.js",
```

Hexcles commented on src/extractScripts.js:
```diff
@@ -0,0 +1,82 @@
+/**
+ * extractScriptsFromHTML extracts the javascript from within <script>
+ * tags in the HTML test provided and writes each script to a .js file.
+ * It creates a directory "tempJsScripts-{tmp numbers}" in the current
```
Sounds like this should be in both `.gitignore` and `.eslintignore`.

dmorejon replied:
_In theory_ the directory is cleaned up always, but yes you're right, thanks!

Hexcles commented on test/testExtractScripts.js:
```diff
@@ -0,0 +1,51 @@
+const assert = require('assert');
```
Nit: use strict

Hexcles reviewed (commented):
LGTM overall.

One more nitpicking suggestion: it's probably better to put both the test inputs and outputs in the same directory underneath `test`, e.g. `test/testdata/{input,reference}`.

LukeZielinski commented on example_legacy_tests/file-list-test.html:
```diff
@@ -0,0 +1,42 @@
+    shouldBeEqualToString("files.item(1).name", "UTF8-2.txt");
+    shouldBeNull("files.item(2)");
+    shouldBeNull("files.item(-1)");
+    shouldBeTrue("files.item(0) === files.item(4294967296)");
```
Can we have a constant for 4294967296 and add/subtract from it below?

dmorejon replied:
This whole file was copy-pasted from a real test in the Chromium source. My intention was not to modify it at all to test my code with some examples of real tests. What do you think, should I still refactor?

LukeZielinski replied:
Ahh, didn't realize that. Maybe throw a comment at the top of the file pointing to where it came from then, and otherwise leave as-is? @Hexcles unless there are issues with c/p that code?

Hexcles replied:
Ahh sorry I missed this. We can add a comment or a README, but the directory name `testdata` should make it obvious that stuff in there are dummy test data.

LukeZielinski commented on src/extractScripts.js:
```diff
@@ -0,0 +1,82 @@
+ * @param {string} filePath - path to the HTML test file to extract
+ * javascript from within <script> tags.
+ * @param {tmp object} [tempDir=null] - OPTIONAL directory to put
+ * new .js files in. If null, will create one.
```
"will create one" - mention where the default will go (ie: TEMP_DIR)

LukeZielinski commented on src/extractScripts.js:
```diff
@@ -0,0 +1,82 @@
+        onopentag(tagname, attribs) {
+          if (tagname === SCRIPT) {
+            inScript = true;
+            const newFileName = '/script' +
```
Should we use platform-specific path separators (instead of "/") here? If so, likely need changes in several files.

dmorejon replied:
We discussed, it seems since this project will run a few times to migrate all the tests and not be run continuously after that, it's okay to assume we're running on Linux for now and if the need arises we will add platform specific separators.

LukeZielinski commented on test/testExtractScripts.js:
```diff
@@ -0,0 +1,51 @@
+
+const extractScriptsFromHTML = require('../src/extractScripts.js');
+
+// IMPORTANT: if file naming convention changes in file, change this function.
```
Maybe add a test that will fail if naming convention doesn't match, and that could be a good place to document what the convention is and which code is affected.

dmorejon replied:
The test in testExtractScripts.js under `it('should return correct list of files', function() {` will fail if the naming convention doesn't match since the list of files returned by the tested function will not be the same as the one generated in `expectedResults()`. I'll add a comment by the `assert` that will highlight this again though.

LukeZielinski commented on test/testExtractScripts.js:
```diff
@@ -0,0 +1,51 @@
+    tempDir.removeCallback();
+  });
+
+  context('2 sets of <script> tags- file-list-test.html', function() {
```
For my own info since I'm new to mocha -- why `context` here instead of `describe`? It sounds like they're equivalent, but is there a preference for using them in a certain way?

dmorejon replied:
My understanding is `context` is an alias of `describe`, [here](https://mochajs.org/#bdd) is the documentation mentioning that. Says that `context` "behaves the same way; it provides a way to keep tests easier to read and organized. Similarly, specify() is an alias for it()"

LukeZielinski reviewed (commented):
meta-nit: would be nice to add a README.md to this repo with things like highlevel description and usage instructions. Doesn't have to be in this pr.

dmorejon committed:
Addressing PR comments https://github.com/Ecosystem-Infra/web-test-transformer/pull/1#issue-441533286

Added some clarification comments and documentation,
moved files to more organized locations,
updated .gitignore and .eslintignore for temp directories,

LukeZielinski reviewed (approved):
Thanks David, LGTM (once we close out the last open comment re: copy-pasting the code)

dmorejon committed:
Changed test expectedResults strings after moving the input and reference files

Hexcles reviewed (approved):
LGTM
